### PR TITLE
Database Refactoring

### PR DESCRIPTION
**Pull request by lcollins11**
Migrated all table-unique CRUD functions to individual table classes and created a decorator file for use within each table class.

### CODE REVIEW

File: src/utils/database/leetcode_question_table.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+from tinydb import TinyDB, where
+from tinydb.table import Document
+from .db_decorators import validate_insert
+
+class LeetcodeQuestionsTable():
+    """
+    Table to hold all questions pulled from Leetcode
+    """
+    TABLE_LEETCODE_QUESTION_FIELDS = ["id", "title", "title_slug", "difficulty"]
+    TABLE_LEETCODE_QUESTION = "Leetcode_Question"
```
Comment (jacobmillsfl) [inline src/utils/database/leetcode_question_table.py]: move to __init__() as self.table_fields and self.table_name

File: src/utils/database/users_table.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+from tinydb import TinyDB, where
+from tinydb.table import Document
+from .db_decorators import validate_insert
+
+class UsersTable():
+    """
+    Table to hold all registered users
+    """
+    TABLE_LEETCODE_USER_FIELDS = ["discord_id", "leetcode_id"]
+    TABLE_LEETCODE_USER = "Leetcode_User"
```
Comment (jacobmillsfl) [inline src/utils/database/users_table.py]: move to __init__() as self.table_fields and self.table_name

File: src/utils/database/weekly_challenge_table.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+from tinydb import TinyDB, where
+from .db_decorators import validate_insert
+
+class WeeklyChallengeTable():
+    """
+    Table to hold all Weekly Challenges after being generated
+    """
+    TABLE_WEEKLY_CHALLENGE_FIELDS = ["id", "date"]
+    TABLE_WEEKLY_CHALLENGE = "Weekly_Challenge"
```
Comment (jacobmillsfl) [inline src/utils/database/weekly_challenge_table.py]: move to __init__() as self.table_fields and self.table_name

File: src/utils/database/weeklyquestion_table.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    A class to act on the weeklyquestion_table of the database
+    """
+    TABLE_WEEKLY_QUESTION_FIELDS = [
+        "id",
+        "challenge_id",
+        "title",
+        "title_slug",
+        "difficulty",
+    ]
+    TABLE_WEEKLY_QUESTION = "Weekly_Question"
```
Comment (jacobmillsfl) [inline src/utils/database/weeklyquestion_table.py]: move to __init__() as self.table_fields and self.table_name

File: src/utils/database/leetcode_question_table.py (hunk 1 above)
Comment (lcollins11) [reply src/utils/database/leetcode_question_table.py]: Renamed but kept outside of constructor due to issues with the validate_insert decorator.

File: src/utils/database/users_table.py (hunk 2 above)
Comment (lcollins11) [reply src/utils/database/users_table.py]: Renamed but kept outside of constructor due to issues with the validate_insert decorator.

File: src/utils/database/weekly_challenge_table.py (hunk 3 above)
Comment (lcollins11) [reply src/utils/database/weekly_challenge_table.py]: Renamed but kept outside of constructor due to issues with the validate_insert decorator.

File: src/utils/database/weeklyquestion_table.py (hunk 4 above)
Comment (lcollins11) [reply src/utils/database/weeklyquestion_table.py]: Renamed but kept outside of constructor due to issues with the validate_insert decorator.

**Review by jacobmillsfl — APPROVED** [review]

**Pull request merged**